### PR TITLE
Upgrade config

### PR DESCRIPTION
**Pull request by nini-faroux**
This adds the initial versions of upgrade config and treasury contracts and tests.

### CODE REVIEW

File: src/Dao/Utils/Value.purs [hunk 1]
```diff
@@ -4,29 +4,75 @@ Description: Value related helpers
...
 import Contract.AssocMap as AssocMap
-import Contract.Prelude (sum, zero, (<$>), (<=<))
+import Contract.Prelude
+  ( class Foldable
+  , type (/\)
+  , all
+  , foldMap
+  , sub
+  , sum
+  , zero
```
Comment (Randy1Burrell) [inline src/Dao/Utils/Value.purs:21]: Why do you import zero twice?

File: src/Dao/Utils/Value.purs [hunk 2]
```diff
@@ -4,29 +4,75 @@ Description: Value related helpers
...
+  , zero
+  , ($)
+  , (/=)
+  , (/\)
+  , (<$>)
+  , (<<<)
+  , (<=<)
+  , (<=<)
```
Comment (Randy1Burrell) [inline src/Dao/Utils/Value.purs]: Why import this twice?

File: test/Workflow/TreasuryTrip.purs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+import Contract.Log (logInfo')
+import Contract.Monad (liftedM)
+import Contract.Prelude
+  ( type (/\)
+  , Unit
+  , bind
+  , discard
+  , pure
+  , show
+  , show
```
Comment (Randy1Burrell) [inline test/Workflow/TreasuryTrip.purs]: `show` is imported multiple times here.

**Review by Randy1Burrell — CHANGES_REQUESTED** [review]
@nini-faroux I left a few comments.

**Comment by nini-faroux** [comment]
@Randy1Burrell - not intentional, just working on it over a period of time, forget that I imported something already, running formatter then puts them beside each other. Thanks for spotting them, removed and will merge now and bring next one up to date.

**Pull request merged**